app/controllers/mixins/actions/host_actions/discover.rb

```diff
@@ -23,6 +23,8 @@ def discover
           elsif session[:type] == "ems"
```

Comment:
Now that you are touching this think (and know what it actually does), could you, please, extract a method here?

Something like `discover_type(type, controller_name)` maybe?